### PR TITLE
Invoice Builder

### PR DESCRIPTION
**Pull request by ishalazo**

### CODE REVIEW

File: src/main/java/com/castletroymedical/billing/InvoiceBuilder.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public Invoice invoiceFactory(String type){
+        if(type == null || type.isEmpty())
+            return null;
+
+        Invoice out;
+        type = type.toLowerCase();
+        switch(type) {
+            case "insured": out = new InsuredInvoice(); break;
+            case "medical card": out = new MedicalCardInvoice(); break;
+            default: out = new PrivateInvoice(); break;
```
Comment (clodaghwalsh17) [inline src/main/java/com/castletroymedical/billing/InvoiceBuilder.java]: Does this mean that any other string provided to the factory will result in a privateInvoice being created?

Comment (ishalazo) [reply src/main/java/com/castletroymedical/billing/InvoiceBuilder.java]: Yes, technically. If we're giving the front side only a choice of 3 which is "insured", "medical card" and "private," there's no other kind of input but those, was my logic for it. I can easily change it out to be private tho, no bother.

Comment (clodaghwalsh17) [reply src/main/java/com/castletroymedical/billing/InvoiceBuilder.java]: I moreso meant have a case for private and the default case be null. Changing it to private would be a good idea too.

**Review by clodaghwalsh17 — APPROVED** [review]
Builder looks good

**Pull request merged**